File: app/src/main/java/com/learning/ruslan/databases/TaskCursorWrapper.java
package com.learning.ruslan.databases;

import android.database.Cursor;
import android.database.CursorWrapper;

import com.learning.ruslan.Assent;
import com.learning.ruslan.Suffix;
import com.learning.ruslan.Task;
import com.learning.ruslan.Word;

public class TaskCursorWrapper extends CursorWrapper {

    public TaskCursorWrapper(Cursor cursor) {
        super(cursor);
    }

    public Assent getAssent() {
        return new Assent(
                getString(getColumnIndex(RusDbSchema.AssentTable.Cols.WORD)),
                getInt(getColumnIndex(RusDbSchema.AssentTable.Cols.POSITION)),
                getInt(getColumnIndex(RusDbSchema.AssentTable.Cols.CHECKED)) == 1,
                getInt(getColumnIndex(RusDbSchema.AssentTable.Cols.ID)));
    }

    public Suffix getSuffix() {
        return new Suffix(
                getString(getColumnIndex(RusDbSchema.SuffixTable.Cols.WORD)),
                getInt(getColumnIndex(RusDbSchema.SuffixTable.Cols.POSITION)),
                getString(getColumnIndex(RusDbSchema.SuffixTable.Cols.ALTERNATIVE)));
    }

    public Paronym getParonym() {
        return new Paronym(
                getString(getColumnIndex(RusDbSchema.ParonymTable.Cols.WORD)),
                getString(getColumnIndex(RusDbSchema.ParonymTable.Cols.ALTERNATIVE)),
                getString(getColumnIndex(RusDbSchema.ParonymTable.Cols.VARIANTS)).split(" ")
        );
    }

    public Task getTask(int typeId) {
        switch (typeId) {
            case Word.AssentId:
                return getAssent();
            case Word.SuffixId:
                return getSuffix();
            case Word.ParonymId:
                return getParonym();
        }
        return null;
    }
}

File: app/src/main/java/com/learning/ruslan/databases/TaskCursorWrapper.kt
package com.learning.ruslan.databases
//
//import android.database.Cursor
//import android.database.CursorWrapper
//import com.learning.ruslan.task.Paronym
//import com.learning.ruslan.task.Suffix
//import com.learning.ruslan.task.Task
//import com.learning.ruslan.WordViewModel
//import com.learning.ruslan.databases.RusDbSchema.*
//import com.learning.ruslan.task.Assent
//
//class TaskCursorWrapper(cursor: Cursor?) : CursorWrapper(cursor) {
//    val assent: Assent
//        get() = Assent(
//            getString(getColumnIndex(AssentTable.Cols.WORD)),
//            getInt(getColumnIndex(AssentTable.Cols.POSITION)),
//            getInt(getColumnIndex(AssentTable.Cols.CHECKED)) == 1,
//            getInt(getColumnIndex(AssentTable.Cols.ID))
//        )
//    val suffix: Suffix
//        get() = Suffix(
//            getString(getColumnIndex(SuffixTable.Cols.WORD)),
//            getInt(getColumnIndex(SuffixTable.Cols.POSITION)),
//            getString(getColumnIndex(SuffixTable.Cols.ALTERNATIVE)))
//    val paronym: Paronym
//        get() = Paronym(
//            getString(getColumnIndex(ParonymTable.Cols.WORD)),
//            getString(getColumnIndex(ParonymTable.Cols.VARIANTS)),
//            getString(getColumnIndex(ParonymTable.Cols.ALTERNATIVE))
//        )
//
//    fun getTask(typeId: Int): Task {
//        return when (typeId) {
//            WordViewModel.Companion.SuffixId -> suffix
//            WordViewModel.Companion.ParonymId -> paronym
//            else -> assent
//        }
//    }
//}